### DISCUSSION TITLE
Terragrunt Support?

### DISCUSSION THREAD

**Discussion by ddvdozuki** [D10S0VSkY-OSS/Stack-Lifecycle-Deployment#92] [Ideas]
Really love this project and I'm hoping we can use it but our entire stack uses terragrunt which means we don't have any vanilla terraform to deploy. It's all provisioned and configured by terragrunt. Is there some way I can modify the stack config to use terragrunt or is this on the roadmap perhaps?

**Comment by D10S0VSkY-OSS** [comment]
Hi @ddvdozuki 
We can discuss it, but the idea of ​​SLD is precisely to improve the DRY of both terraform and terragrunt.
Keep in mind that with SLD it is only necessary to use a stack that makes use of one or more modules and you can pass the tfvars to it for each environment or account without having to repeat anything more than the parameters of the tfvars files.

**Reply by ddvdozuki** [reply]
It kinda seems like this would replace the need for terragrunt if thats the case. You'd be building a whole new set of dependencies with SLD as the dependency manager and deployer right? Maybe I'm expecting SLD to do too much. I was kinda hoping I could use it to allow my team and other less-technical people to deploy our stack at will for testing, qa, pilots, etc. But I'm not really willing to re-write my entire infrastructure or how it's deployed to get that convenience. For the record, you can look at our repo here: [(https://github.com/Dozuki/CloudPrem-Config)](https://github.com/Dozuki/CloudPrem-Config) this is the config and the entry point to the infra, you would run `terragrunt run-all apply` from within one of the environment folders (the ones that have an `env.hcl` file). Here is a link to the terraform itself: https://github.com/Dozuki/CloudPrem-Infra/tree/master/cloudprem

**Comment by ramazulay** [comment]
Hi @D10S0VSkY-OSS ,
Any update on terragrunt support?

**Reply by D10S0VSkY-OSS** [reply]
@ddvdozuki @ramazulay 
Terragrun is added with the basic commands of plan, apply, and destroy as a Beta version
[v3.3.0](https://github.com/D10S0VSkY-OSS/Stack-Lifecycle-Deployment/releases/tag/v3.3.0)
